ModuleNotFoundError: No module named 'test' in tests/test_impl.py

**Describe the bug**
Using a `python:3.11.7` docker container I run into the following import error when running the test suite.

```pytb
ImportError while importing test module '/root/aiohappyeyeballs/tests/test_impl.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/local/lib/python3.11/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_impl.py:3: in <module>
    from test.test_asyncio import utils as test_utils
E   ModuleNotFoundError: No module named 'test'
```

**To Reproduce**
Steps to reproduce the behavior:

```sh
git clone https://github.com/bdraco/aiohappyeyeballs
cd aiohappyeyeballs/
poetry install
poetry run pytest
```

**Additional context**
The `test` module is documented for internal use by the python interpreter and apparently not present in all Python distributions.

> The [test](https://docs.python.org//3.12/library/test.html#module-test) package is meant for internal use by Python only. It is documented for the benefit of the core developers of Python. Any use of this package outside of Python’s standard library is discouraged as code mentioned here can change or be removed without notice between releases of Python.

https://docs.python.org//3.12/library/test.html#module-test

--- a/tests/test_impl.py
+++ b/tests/test_impl.py
@@ -1,6 +1,5 @@
 import asyncio
 import socket
-from test.test_asyncio import utils as test_utils
 from types import ModuleType
 from typing import Tuple
 from unittest import mock
@@ -30,9 +29,21 @@ def mock_socket_module():
             delattr(m_socket, name)
 
     m_socket.socket = mock.MagicMock()
-    m_socket.socket.return_value = test_utils.mock_nonblocking_socket()
+    m_socket.socket.return_value = mock_nonblocking_socket()
 
     return m_socket
+
+
+def mock_nonblocking_socket(
+    proto=socket.IPPROTO_TCP, type=socket.SOCK_STREAM, family=socket.AF_INET
+):
+    """Create a mock of a non-blocking socket."""
+    sock = mock.create_autospec(socket.socket, spec_set=True, instance=True)
+    sock.proto = proto
+    sock.type = type
+    sock.family = family
+    sock.gettimeout.return_value = 0.0
+    return sock
 
 
 def patch_socket(f):

